pkg/integrations/prometheus/utils/operatorutils.go

```diff
@@ -0,0 +1,215 @@
+		return segutils.LetAnd
+	case parser.LOR:
+		return segutils.LetOr
+	case parser.LUNLESS:
+		return segutils.LetUnless
+	default:
+		log.Errorf("getArithmeticOperation: unexpected op: %v", op)
+		return 0
+	}
+}
+
+func IsLogicalOperator(op segutils.LogicalAndArithmeticOperator) bool {
+	switch op {
+	case segutils.LetAnd:
+		return true
+	case segutils.LetOr:
+		return true
+	case segutils.LetUnless:
+		return true
+	default:
+		return false
+	}
+}
+
+func SetFinalResult(queryOp structs.QueryArithmetic, finalResult map[string]map[uint32]float64, groupID string, timestamp uint32, valueLHS float64, valueRHS float64, swapped bool) {
+
+	switch queryOp.Operation {
+	case utils.LetAdd:
+		finalResult[groupID][timestamp] = valueLHS + valueRHS
+	case utils.LetDivide:
+		if valueRHS == 0 {
+			return
+		}
+		if swapped {
+			finalResult[groupID][timestamp] = valueRHS / valueLHS
+		} else {
+			finalResult[groupID][timestamp] = valueLHS / valueRHS
+		}
+	case utils.LetMultiply:
+		finalResult[groupID][timestamp] = valueLHS * valueRHS
+	case utils.LetSubtract:
+		val := valueLHS - valueRHS
+		if swapped {
+			val = val * -1
+		}
+		finalResult[groupID][timestamp] = val
+	case utils.LetModulo:
+		if swapped {
+			finalResult[groupID][timestamp] = math.Mod(valueRHS, valueLHS)
+		} else {
+			finalResult[groupID][timestamp] = math.Mod(valueLHS, valueRHS)
+		}
+	case utils.LetPower:
+		if swapped {
+			finalResult[groupID][timestamp] = math.Pow(valueRHS, valueLHS)
+		} else {
+			finalResult[groupID][timestamp] = math.Pow(valueLHS, valueRHS)
+		}
+	case utils.LetGreaterThan:
+		isGtr := valueLHS > valueRHS
+		if swapped {
+			isGtr = valueLHS < valueRHS
+		}
+		if isGtr {
+			finalResult[groupID][timestamp] = valueLHS
+		}
+	case utils.LetGreaterThanOrEqualTo:
+		isGte := valueLHS >= valueRHS
+		if swapped {
+			isGte = valueLHS <= valueRHS
+		}
+		if isGte {
+			finalResult[groupID][timestamp] = valueLHS
+		}
+	case utils.LetLessThan:
+		isLss := valueLHS < valueRHS
+		if swapped {
+			isLss = valueLHS > valueRHS
+		}
+		if isLss {
+			finalResult[groupID][timestamp] = valueLHS
+		}
+	case utils.LetLessThanOrEqualTo:
+		isLte := valueLHS <= valueRHS
+		if swapped {
+			isLte = valueLHS >= valueRHS
+		}
+		if isLte {
+			finalResult[groupID][timestamp] = valueLHS
+		}
+	case utils.LetEquals:
+		if valueLHS == valueRHS {
+			finalResult[groupID][timestamp] = valueLHS
+		}
+	case utils.LetNotEquals:
+		if valueLHS != valueRHS {
+			finalResult[groupID][timestamp] = valueLHS
+		}
+	}
+
+	// Logical ops can only been used between 2 vectors
+	if !queryOp.ConstantOp && IsLogicalOperator(queryOp.Operation) {
+		switch queryOp.Operation {
+		case utils.LetAnd:
+			finalResult[groupID][timestamp] = valueLHS
+		case utils.LetOr:
+			finalResult[groupID][timestamp] = valueLHS
+		case utils.LetUnless:
+			finalResult[groupID][timestamp] = valueLHS
+		}
+	}
+}
+
+func ExtractMatchingLabelSet(groupIDStr string, matchingLabels []string, on bool) string {
```

Comment:
Can you add a comment above this function giving a simple example of how it's supposed to work? Later if someone else wants to use this function (or quickly check if this function can be used for what they want to do), it would be great if they can understand how it works from a simple example instead of having to read quite a bit of code